processors/enduserprocessor/enduserprocessor.go

```diff
@@ -0,0 +1,534 @@
+package enduserprocessor
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/PaesslerAG/jsonpath"
+	"github.com/dgrijalva/jwt-go"
+	jsoniter "github.com/json-iterator/go"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+	"go.uber.org/zap"
+
+	"github.com/hypertrace/collector/processors/enduserprocessor/hash"
+	"github.com/hypertrace/collector/processors/piifilterprocessor"
+)
+
+const (
+	enduserIDAttribute      = "enduser.id"
+	enduserRoleAttribute    = "enduser.role"
+	enduserScopeAttribute   = "enduser.scope"
+	enduserSessionAttribute = "session.id"
+)
+
+type config struct {
+	EndUser
+	hashAlgorithm hash.Algorithm
```

Comment:
Not sure this is still a thing maybe @davexroth can confirm? In PII filter we only use hash.